File: packages/lib-config/src/database/database.models.ts
import { type DatabaseTypeModel } from '@lib/backend/database/database.models';
import { type ClassModel } from '@lib/shared/core/core.models';
import { type EntityResourceModel } from '@lib/shared/resource/resources/EntityResource/EntityResource.models';
import { type Options } from '@mikro-orm/core/utils';
import { type MongoDriver } from '@mikro-orm/mongodb';

export type DatabaseConfigModel = {
  database?: string;
  entities: Array<ClassModel<EntityResourceModel>>;
  expireSeconds: number;
  host?: string;
  password?: string;
  pool: { max: number };
  type?: DatabaseTypeModel;
  username?: string;
};

export type _DatabaseConfigModel = Options<MongoDriver>;

File: packages/lib-config-py/src/lib_config/database/database_models.py
from typing import Any

from lib_shared.core.utils.base_model import BaseModel
from lib_shared.core.utils.field.field import Field


class DatabaseConfigModel(BaseModel):
    database: str = Field()
    resources: list[Any] = []
    host: str = Field()
    password: str = Field()
    username: str = Field()
    max_pool: int = Field()
    min_pool: int = Field()
    timeout: int = Field()
